{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "7e473b27-53e5-46f9-9fe3-164974765964",
   "metadata": {},
   "source": [
    "# Run FIRST"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "c5ae8013-a502-41bd-b538-e586d35d2ecd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of raw samples:  0\n",
      "Number of clean samples:  1080\n",
      "Number of assembles:  1446\n"
     ]
    }
   ],
   "source": [
    "### VARIABLES ###\n",
    "\n",
    "#python variables\n",
    "REF_GENS = {\n",
    "    'Chicken': '/beegfs/work/graaf20/Databases/Ref_gens/Chicken/GCA_024206055.2_GGswu_genomic.fna',\n",
    "    'Corn': '/beegfs/work/graaf20/Databases/Ref_gens/Corn/GCA_902167145.1_Zm-B73-REFERENCE-NAM-5.0_genomic.fna'\n",
    "}\n",
    "\n",
    "\n",
    "#shared between bash and python\n",
    "VARS = (\n",
    "    'WS=/beegfs/work/graaf20/ChickenPhages\\n'\n",
    "    'DB=/beegfs/work/graaf20/Databases/\\n'\n",
    "    'DATA=Data\\n'\n",
    "    'RES=Results\\n'\n",
    "    'ASS=Results/Assemlies_MegaHit\\n'\n",
    "    'MH=MegaHit_temp\\n'\n",
    "    'FIG=Figures\\n'\n",
    "    'LOGS=Logs\\n'\n",
    "    'RR=Raw_reads\\n'\n",
    "    'CR=Clean_reads\\n'\n",
    ")\n",
    "\n",
    "#METADATA = pd.read_csv('metadata.tsv', sep='\\t', index_col=0)\n",
    "\n",
    "CONDA = '$HOME/miniconda3/etc/profile.d/conda.sh'\n",
    "\n",
    "\n",
    "##################### DO NOT CHANGE ################################\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import os\n",
    "\n",
    "#extract bash variables for python\n",
    "for var in VARS.strip().split('\\n'):\n",
    "    vars()[var.split('=')[0]]=var.split('=')[1]\n",
    "\n",
    "!mkdir -p $WS/$CR $WS/$LOGS $WS/$TABS $WS/$RES $WS/$FIG\n",
    "    \n",
    "#Binac function\n",
    "params = '''\n",
    "#PBS -l nodes={nodes}:ppn={ppn}\n",
    "#PBS -l walltime={time}\n",
    "#PBS -l mem={mem}gb\n",
    "#PBS -N {name}.sh\n",
    "#PBS -o {logdir}\n",
    "#PBS -e {logdir}\n",
    "#PBS -q {queue}\n",
    "#PBS -S /bin/bash\n",
    "export TMPDIR=$TMPDIR\n",
    "''' \n",
    "\n",
    "source = f'source {CONDA}'\n",
    "\n",
    "COMM = '#Custom variables and commands\\n{todo}'\n",
    "STRING = '\\n'.join([params, source, VARS, COMM])\n",
    "\n",
    "# binac function that will submit the job to server\n",
    "def binac(nodes=1, ppn=1, time='00:15:00', mem=2, name='myjob', logdir=WS,\n",
    "          queue='tiny', todo='echo \"Command\"', STRING=STRING, ):\n",
    "    text = STRING\n",
    "    if '{nodes}' in STRING:\n",
    "        text = STRING.format(nodes=nodes, ppn=ppn, time=time, logdir=f'{WS}/{logdir}',\n",
    "               mem=mem, name=name, queue=queue, todo=todo)\n",
    "    script = f'{WS}/{logdir}/{name}.sh'\n",
    "    !mkdir -p $WS/$logdir\n",
    "    with open(script, \"w\") as text_file:\n",
    "        text_file.write(text)\n",
    "    !chmod +x $script\n",
    "    !qsub $script\n",
    "    \n",
    "###############################################################################################\n",
    "\n",
    "bad = ['SRR5280393', 'ERR2241736', 'ERR2241917', 'SRR12730717', 'SRR7683044',\n",
    "       'SRR7457769', 'SRR9113694', 'ERR2241801', 'SRR17836573', 'SRR7685268', \n",
    "       'SRR9113697', 'SRR9113700', 'SRR5280759', 'SRR17836591', 'SRR7457773', \n",
    "       'SRR7683048', 'ERR2241740', 'ERR2241741', 'SRR17836604', 'SRR7683050', \n",
    "       'SRR7457774', 'ERR2241809', 'SRR5282097', 'SRR12730726', 'SRR7683052',\n",
    "       'ERR2241763', 'ERR2241943', 'SRR7685282', 'SRR17836559', 'SRR7457783',\n",
    "       'ERR2241850', 'ERR2241735', 'ERR2241764', 'SRR12730743', 'SRR7457734', \n",
    "       'ERR2241631', 'SRR7683056', 'SRR7457738', 'SRR7683059',  'ERR2241635',\n",
    "       'SRR7457741', 'ERR2241791', 'ERR2241795', 'SRR7457745',  'ERR2241689', \n",
    "       'SRR7685255', 'SRR7685256', 'ERR2241698', 'SRR7457755', ] \n",
    "\n",
    "# get batches and locals\n",
    "df = pd.read_csv(f'{WS}/Data/MAGs_dataset_complete.csv', index_col=0, na_filter = False)\n",
    "df.BioProject = df.BioProject.str.replace('NA', 'Local')\n",
    "#locs = df.loc[df.BioProject == 'Local', 'Run'].unique().tolist()\n",
    "#sras = df.loc[df.BioProject != 'Local', 'Run'].unique().tolist()\n",
    "sras = df.loc[~df.Run.isin(bad), 'Run'].unique().tolist()\n",
    "b = 30\n",
    "batches = {f'batch{i+1}-{i+b}': sras[i:i + b] for i in range(0, len(sras), b)}\n",
    "\n",
    "# Create list of file indexes to access it later\n",
    "cleaning = !rm -r $WS/*/*/*/.ipynb_ch* $WS/*/*/.ipynb_ch* $WS/*/.ipynb_ch* $WS/.ipynb_ch*\n",
    "\n",
    "raws = []\n",
    "if os.path.exists(f'{WS}/{RR}'): \n",
    "    raws = [f.rsplit('_', 1)[0] for f in os.listdir(f'{WS}/{RR}') if '.gz' in f]\n",
    "    raws = sorted(list(set(raws)))\n",
    "    print('Number of raw samples: ', len(raws))\n",
    "\n",
    "cleans = []\n",
    "if os.path.exists(f'{WS}/{CR}'): \n",
    "    cleans = [f.rsplit('_', 1)[0] for f in os.listdir(f'{WS}/{CR}') if '.gz' in f]\n",
    "    cleans = sorted(list(set(cleans)))\n",
    "    print('Number of clean samples: ', len(cleans))\n",
    "    \n",
    "asses = []\n",
    "if os.path.exists(f'{WS}/{ASS}'): \n",
    "    asses = [f.rsplit('.', 1)[0] for f in os.listdir(f'{WS}/{ASS}') if '.fa' in f]\n",
    "    asses = sorted(list(set(asses)))\n",
    "    print('Number of assembles: ', len(asses))\n",
    "    \n",
    "permissions = !chmod +rwx $BIN/*\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "28efb06f-7e1a-4388-a0c9-d9bd861ccc51",
   "metadata": {},
   "source": [
    "# SRA-Toolkit"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "id": "b0eea95e-3025-413d-92f7-873d5456b249",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11366442\n"
     ]
    }
   ],
   "source": [
    "#use only 1 thread or bad things may happen (Binac will freeze)!\n",
    "\n",
    "THREADS = 2\n",
    "logdir = f'{LOGS}/SRAs'\n",
    "\n",
    "todo = '''\n",
    "conda activate QC\n",
    "\n",
    "cd $WS\n",
    "mkdir -p sra_temp\n",
    "\n",
    "echo \"Processing {BATCH}\"\n",
    "sras=({IDS})\n",
    "for sra in ${{sras[@]}}; do\n",
    "\n",
    "    #skip processed files\n",
    "    if [ -f $RR/${{sra}}_1.fastq.gz ]; then\n",
    "        echo \"$sra already exists. Skipping.\"\n",
    "        continue\n",
    "    fi\n",
    "    \n",
    "    if [ -f $RR/${{sra}}.fastq.gz ]; then\n",
    "        echo \"$sra already exists. Skipping.\"\n",
    "        continue\n",
    "    fi\n",
    "    \n",
    "    if [ -f $CR/${{sra}}_1.fq.gz ]; then\n",
    "        echo \"$sra already exists. Skipping.\"\n",
    "        continue\n",
    "    fi\n",
    "    \n",
    "    if [ -f $CR/${{sra}}.fq.gz ]; then\n",
    "        echo \"$sra already exists. Skipping.\"\n",
    "        continue\n",
    "    fi\n",
    "    \n",
    "    #download files\n",
    "    echo \"Prefetch $sra\"\n",
    "    prefetch $sra -O sra_temp --verify yes --max-size 50G\n",
    "\n",
    "    echo \"Convert to fastq $sra\"\n",
    "    fasterq-dump sra_temp/$sra --split-files --outdir $RR --threads 1\n",
    "    rm -r sra_temp/$sra\n",
    "\n",
    "    echo \"Gzip $sra\"\n",
    "    pigz $RR/${{sra}}_*.fastq\n",
    "    pigz $RR/${{sra}}.fastq\n",
    "\n",
    "    \n",
    "done\n",
    "echo \"All done, check the outputs\"\n",
    "'''\n",
    "\n",
    "!mkdir -p $WS/$RR\n",
    "\n",
    "for batch, ids in batches.items():\n",
    "    if batch not in ['batch1001-1020']:\n",
    "        continue\n",
    "        \n",
    "    IDS = ' '.join([f\"'{l}'\" for l in [i for i in ids if i not in bad]])\n",
    "    binac(ppn=THREADS, time='07:00:00:00', mem=16, name=f'{batch}_SRA', queue='long', \n",
    "          logdir=logdir, todo=todo.format(BATCH=batch, IDS=IDS))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "fb43fca3-edb1-43bc-aa3f-1cba585e2967",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Summarize raw data\n",
    "\n",
    "df['Status'] = 'Downloaded'\n",
    "df.loc[df.Run.isin(bad), 'Status'] = 'Broken'\n",
    "\n",
    "df.to_csv(f'{WS}/Data/MAGs_dataset_complete.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "9f403cde-fff4-4fe9-8ca7-bc5c909eb683",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1515"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(df.loc[df.Status == 'Downloaded'])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b5956f53-194a-4a4e-bd96-d9841613c3f5",
   "metadata": {},
   "source": [
    "# Local samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e2119a5d-6bd5-426c-9d0a-dc1b08ef21da",
   "metadata": {},
   "outputs": [],
   "source": [
    "#copy assembled contigs for UHO samples and raw data for LH samples\n",
    "\n",
    "!mkdir -p $WS/$ASS\n",
    "\n",
    "for loc in locs:\n",
    "    if loc.startswith('UHO'):\n",
    "        continue\n",
    "        src = f'~/Beeg/PFOWL2-3/Results/MegaHit/{loc}/final.contigs.fa'\n",
    "        new = f'{WS}/{ASS}/{loc}.fa'\n",
    "        \n",
    "        !cp $src $new\n",
    "        \n",
    "    if loc.startswith('LH'):\n",
    "        for i in range(1,3):\n",
    "            src = f'/beegfs/work/croth/LH1_metagenomes/Clean_reads/{loc}_{i}.fq.gz'\n",
    "            new = f'{WS}/{RR}/{loc}_{i}.fastq.gz'\n",
    "            if os.path.exists(new) or os.path.exists(f'{WS}/{CR}/{loc}_{i}.fq.gz'):\n",
    "                continue\n",
    "            \n",
    "            !cp $src $new"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6c303b34-2e1a-481e-bbd7-c425df273aac",
   "metadata": {
    "tags": []
   },
   "source": [
    "# ReadQC and host removal"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "69d11bed-b56b-430d-902d-887dfdd7e48d",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11130446\n",
      "11130447\n"
     ]
    }
   ],
   "source": [
    "#create index\n",
    "\n",
    "THREADS = 8\n",
    "\n",
    "todo = '''\n",
    "conda activate QC\n",
    "\n",
    "cd {ws}\n",
    "bowtie2-build {refer} {index} --threads {THREADS}\n",
    "'''\n",
    "\n",
    "for ind, ref in REF_GENS.items():\n",
    "    ws,fa = ref.rsplit('/', 1)\n",
    "    binac(ppn=THREADS, time='08:00:00', mem=90, name=f'{ind}_Ref', queue='short', \n",
    "          logdir=LOGS, todo=todo.format(ws=ws, refer=fa, index=ind, THREADS=THREADS))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "id": "4412a159-652a-40e3-b3d7-9b48a023752a",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11366477\n"
     ]
    }
   ],
   "source": [
    "#Read QC and host DNA removal\n",
    "\n",
    "THREADS = 8\n",
    "logdir = f\"{LOGS}/QC\"\n",
    "\n",
    "todo = '''\n",
    "conda activate QC\n",
    "\n",
    "echo \"Processing {BATCH}\"\n",
    "sras=({IDS})\n",
    "for name in ${{sras[@]}}; do\n",
    "    cd $WS\n",
    "    ref_host={ref_host}\n",
    "    ind_host={ind_host}\n",
    "    ref_feed={ref_feed}\n",
    "    ind_feed={ind_feed}\n",
    "    preQC=$DATA/FastQC/preQC/$name\n",
    "    postQC=$DATA/FastQC/postQC/$name\n",
    "    outdir=$CR/$name\n",
    "    \n",
    "    echo\n",
    "    echo \"############################################################\"\n",
    "    echo \"Processing $name \"\n",
    "    echo \"Check if reads are paired or single\" \n",
    "    \n",
    "    if [ -f $RR/${{name}}_1.f*q.gz ]; then\n",
    "        echo \"$name are paired reads.\"\n",
    "        RR1=$RR/${{name}}_1.f*q.gz\n",
    "        RR2=$RR/${{name}}_2.f*q.gz\n",
    "        CR1=$CR/${{name}}_1.fq.gz\n",
    "        CR2=$CR/${{name}}_2.fq.gz\n",
    "    \n",
    "        #skip processed files\n",
    "        if [ -f $CR1 ]; then\n",
    "            echo \"$name already cleaned. Skipping.\"\n",
    "            continue\n",
    "        fi\n",
    "        \n",
    "        rm -rf $outdir $preQC $postQC $CR1 $CR2\n",
    "        mkdir -p $preQC $postQC $outdir \n",
    "    \n",
    "        # Fastq report before QC\n",
    "        echo \"Report before QC...\"\n",
    "        fastqc -q -t {THREADS} -o $preQC -f fastq $RR1 $RR2\n",
    "        \n",
    "        echo\n",
    "        echo \"Running Trim-Galore for QC\"\n",
    "        echo   \n",
    "        cd $outdir\n",
    "        trim_galore --paired $WS/$RR1 $WS/$RR2 -j {THREADS}\n",
    "    \n",
    "        # Fastq report after QC\n",
    "        echo \"Report after QC...\"\n",
    "        cd\n",
    "        cd $WS\n",
    "        fastqc -q -t {THREADS} -o $postQC -f fastq $outdir/*_val_1.f*q.gz $outdir/*_val_2.f*q.gz\n",
    "        \n",
    "        echo\n",
    "        echo \"Removing Host and Feed DNA from reads\"\n",
    "        echo \"Mapping reads to the host reference...\"\n",
    "        cd\n",
    "        cd $ref_host\n",
    "        bowtie2 -p {THREADS} -x $ind_host -1 $WS/$outdir/*_val_1.f*q.gz -2 $WS/$outdir/*_val_2.f*q.gz \\\n",
    "        --un-conc-gz $WS/$outdir/no_host > $WS/$outdir/host.sam\n",
    "        echo \"Host DNA removed\"\n",
    "        \n",
    "        echo \"Mapping reads to the feed reference...\"\n",
    "        cd\n",
    "        cd $ref_feed\n",
    "        bowtie2 -p {THREADS} -x $ind_feed -1 $WS/$outdir/no_host.1 -2 $WS/$outdir/no_host.2 \\\n",
    "        --un-conc-gz $WS/$outdir/no_feed > $WS/$outdir/feed.sam\n",
    "        echo \"Feed DNA removed\"\n",
    "        \n",
    "        echo\n",
    "        echo \"Move and clean!\"\n",
    "        cd\n",
    "        cd $WS\n",
    "        mv $outdir/no_feed.1 $CR1\n",
    "        mv $outdir/no_feed.2 $CR2\n",
    "        rm -rf $outdir\n",
    "    \n",
    "        #remove processed files\n",
    "        if [ -f $CR1 ]; then\n",
    "            echo \"Removing raw read since $name successfully cleaned.\"\n",
    "            rm $RR1 $RR2\n",
    "        fi\n",
    "    fi\n",
    "\n",
    "    if [ -f $RR/${{name}}.f*q.gz ]; then\n",
    "        echo \"$name are single reads.\"\n",
    "        RR1=$RR/${{name}}.f*q.gz\n",
    "        CR1=$CR/${{name}}.fq.gz\n",
    "    \n",
    "        #skip processed files\n",
    "        if [ -f $CR1 ]; then\n",
    "            echo \"$name already cleaned. Skipping.\"\n",
    "            continue\n",
    "        fi\n",
    "        \n",
    "        rm -rf $outdir $preQC $postQC $CR1 $CR2\n",
    "        mkdir -p $preQC $postQC $outdir \n",
    "    \n",
    "        # Fastq report before QC\n",
    "        echo \"Report before QC...\"\n",
    "        fastqc -q -t {THREADS} -o $preQC -f fastq $RR1\n",
    "        \n",
    "        echo\n",
    "        echo \"Running Trim-Galore for QC\"\n",
    "        echo   \n",
    "        cd $outdir\n",
    "        trim_galore $WS/$RR1 -j {THREADS}\n",
    "    \n",
    "        # Fastq report after QC\n",
    "        echo \"Report after QC...\"\n",
    "        cd\n",
    "        cd $WS\n",
    "        fastqc -q -t {THREADS} -o $postQC -f fastq $outdir/*_trimmed.f*q.gz\n",
    "        \n",
    "        echo\n",
    "        echo \"Removing Host and Feed DNA from reads\"\n",
    "        echo \"Mapping reads to the host reference...\"\n",
    "        cd\n",
    "        cd $ref_host\n",
    "        bowtie2 -p {THREADS} -x $ind_host -U $WS/$outdir/*_trimmed.f*q.gz \\\n",
    "        --un-gz $WS/$outdir/no_host > $WS/$outdir/host.sam\n",
    "        echo \"Host DNA removed\"\n",
    "        \n",
    "        echo \"Mapping reads to the feed reference...\"\n",
    "        cd\n",
    "        cd $ref_feed\n",
    "        bowtie2 -p {THREADS} -x $ind_feed -U $WS/$outdir/no_host \\\n",
    "        --un-gz $WS/$outdir/no_feed > $WS/$outdir/feed.sam\n",
    "        echo \"Feed DNA removed\"\n",
    "        \n",
    "        echo\n",
    "        echo \"Move and clean!\"\n",
    "        cd\n",
    "        cd $WS\n",
    "        mv $outdir/no_feed $CR1\n",
    "        rm -rf $outdir\n",
    "    \n",
    "        #remove processed files\n",
    "        if [ -f $CR1 ]; then\n",
    "            echo \"Removing raw read since $name successfully cleaned.\"\n",
    "            rm $RR1\n",
    "        fi\n",
    "        \n",
    "    else\n",
    "        Echo \"There is something wrong with that sample (neither paired reads or single read were found)\"\n",
    "        continue\n",
    "        \n",
    "    fi\n",
    "done\n",
    "\n",
    "echo\n",
    "echo \"############################################################\"\n",
    "echo \"###                   All done! Enjoy!                   ###\"\n",
    "echo \"### Citations:                                           ###\"\n",
    "echo \"### 1. Cutadapt: https://doi.org/10.14806/ej.17.1.200    ###\"\n",
    "echo \"### 2. Bowtie: https://doi.org/10.1038/nmeth.1923        ###\"\n",
    "echo \"### 3. FastQC (optional):                                ###\"\n",
    "echo \"###    https://doi.org/10.1038/nmeth.1923                ###\"\n",
    "echo \"### 4. TrimGalore (optional):                            ###\"\n",
    "echo \"###    https://github.com/FelixKrueger/TrimGalore        ###\"\n",
    "echo \"############################################################\"\n",
    "'''\n",
    "\n",
    "#get indexes for Host/Feed genomes\n",
    "inds, refs = [], []\n",
    "for ind, ref in REF_GENS.items():\n",
    "    inds.append(ind)\n",
    "    refs.append(ref.rsplit('/', 1)[0])\n",
    "\n",
    "\n",
    "#launch samples\n",
    "for batch, ids in batches.items():\n",
    "    if batch not in ['batch1001-1020']:\n",
    "        continue\n",
    "        \n",
    "    IDS = ' '.join([f\"'{l}'\" for l in [i for i in ids if i not in bad]])\n",
    "            \n",
    "    binac(ppn=THREADS, time='07:00:00:00', mem=120, name=f'{batch}_QC', queue='long', \n",
    "              logdir=logdir, todo=todo.format(BATCH=batch, IDS=IDS, ref_host=refs[0], \n",
    "              ind_host=inds[0], ref_feed=refs[1], ind_feed=inds[1], THREADS=THREADS))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7f0a42d0-ac46-48bd-b517-9f6683f68374",
   "metadata": {},
   "outputs": [],
   "source": [
    "#keep output reports \n",
    "\n",
    "logs = f\"{WS}/{LOGS}/QC\"\n",
    "\n",
    "!mkdir {WS}/$logs\n",
    "!mv *_QC.sh.* {WS}/$logs/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "eab06f4c-9069-4a96-bf0a-a04e5129c2c5",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Total</th>\n",
       "      <th>PassedQC</th>\n",
       "      <th>NoHost</th>\n",
       "      <th>NoFeed</th>\n",
       "      <th>PassedQC(%)</th>\n",
       "      <th>Host(%_of_PassedQC)</th>\n",
       "      <th>Feed(%_of_PassedQC)</th>\n",
       "      <th>Owner</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SampleID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>UHO1</th>\n",
       "      <td>35831867.0</td>\n",
       "      <td>35492845.0</td>\n",
       "      <td>34944907.0</td>\n",
       "      <td>34942341.0</td>\n",
       "      <td>99.054</td>\n",
       "      <td>1.544</td>\n",
       "      <td>0.007</td>\n",
       "      <td>Money</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>UHO10</th>\n",
       "      <td>33592806.0</td>\n",
       "      <td>33285677.0</td>\n",
       "      <td>33181082.0</td>\n",
       "      <td>33169124.0</td>\n",
       "      <td>99.086</td>\n",
       "      <td>0.314</td>\n",
       "      <td>0.036</td>\n",
       "      <td>Money</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>UHO100</th>\n",
       "      <td>35201028.0</td>\n",
       "      <td>35184245.0</td>\n",
       "      <td>35159077.0</td>\n",
       "      <td>35156321.0</td>\n",
       "      <td>99.952</td>\n",
       "      <td>0.072</td>\n",
       "      <td>0.008</td>\n",
       "      <td>Money</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>UHO101</th>\n",
       "      <td>35715119.0</td>\n",
       "      <td>35434487.0</td>\n",
       "      <td>35373052.0</td>\n",
       "      <td>35258757.0</td>\n",
       "      <td>99.214</td>\n",
       "      <td>0.173</td>\n",
       "      <td>0.323</td>\n",
       "      <td>Money</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>UHO102</th>\n",
       "      <td>33628784.0</td>\n",
       "      <td>33391514.0</td>\n",
       "      <td>33276130.0</td>\n",
       "      <td>33272830.0</td>\n",
       "      <td>99.294</td>\n",
       "      <td>0.346</td>\n",
       "      <td>0.010</td>\n",
       "      <td>Money</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>UHO95</th>\n",
       "      <td>34632053.0</td>\n",
       "      <td>34323601.0</td>\n",
       "      <td>34297988.0</td>\n",
       "      <td>34286021.0</td>\n",
       "      <td>99.109</td>\n",
       "      <td>0.075</td>\n",
       "      <td>0.035</td>\n",
       "      <td>Money</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>UHO96</th>\n",
       "      <td>41129500.0</td>\n",
       "      <td>40722799.0</td>\n",
       "      <td>40715886.0</td>\n",
       "      <td>40704007.0</td>\n",
       "      <td>99.011</td>\n",
       "      <td>0.017</td>\n",
       "      <td>0.029</td>\n",
       "      <td>Money</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>UHO97</th>\n",
       "      <td>41883780.0</td>\n",
       "      <td>41511029.0</td>\n",
       "      <td>41507829.0</td>\n",
       "      <td>41505113.0</td>\n",
       "      <td>99.110</td>\n",
       "      <td>0.008</td>\n",
       "      <td>0.007</td>\n",
       "      <td>Money</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>UHO98</th>\n",
       "      <td>40138802.0</td>\n",
       "      <td>39775971.0</td>\n",
       "      <td>32287226.0</td>\n",
       "      <td>32230779.0</td>\n",
       "      <td>99.096</td>\n",
       "      <td>18.827</td>\n",
       "      <td>0.142</td>\n",
       "      <td>Money</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>UHO99</th>\n",
       "      <td>67583716.0</td>\n",
       "      <td>67074144.0</td>\n",
       "      <td>66918180.0</td>\n",
       "      <td>66758326.0</td>\n",
       "      <td>99.246</td>\n",
       "      <td>0.233</td>\n",
       "      <td>0.238</td>\n",
       "      <td>Money</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>440 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               Total    PassedQC      NoHost      NoFeed  PassedQC(%)   \n",
       "SampleID                                                                \n",
       "UHO1      35831867.0  35492845.0  34944907.0  34942341.0       99.054  \\\n",
       "UHO10     33592806.0  33285677.0  33181082.0  33169124.0       99.086   \n",
       "UHO100    35201028.0  35184245.0  35159077.0  35156321.0       99.952   \n",
       "UHO101    35715119.0  35434487.0  35373052.0  35258757.0       99.214   \n",
       "UHO102    33628784.0  33391514.0  33276130.0  33272830.0       99.294   \n",
       "...              ...         ...         ...         ...          ...   \n",
       "UHO95     34632053.0  34323601.0  34297988.0  34286021.0       99.109   \n",
       "UHO96     41129500.0  40722799.0  40715886.0  40704007.0       99.011   \n",
       "UHO97     41883780.0  41511029.0  41507829.0  41505113.0       99.110   \n",
       "UHO98     40138802.0  39775971.0  32287226.0  32230779.0       99.096   \n",
       "UHO99     67583716.0  67074144.0  66918180.0  66758326.0       99.246   \n",
       "\n",
       "          Host(%_of_PassedQC)  Feed(%_of_PassedQC)  Owner  \n",
       "SampleID                                                   \n",
       "UHO1                    1.544                0.007  Money  \n",
       "UHO10                   0.314                0.036  Money  \n",
       "UHO100                  0.072                0.008  Money  \n",
       "UHO101                  0.173                0.323  Money  \n",
       "UHO102                  0.346                0.010  Money  \n",
       "...                       ...                  ...    ...  \n",
       "UHO95                   0.075                0.035  Money  \n",
       "UHO96                   0.017                0.029  Money  \n",
       "UHO97                   0.008                0.007  Money  \n",
       "UHO98                  18.827                0.142  Money  \n",
       "UHO99                   0.233                0.238  Money  \n",
       "\n",
       "[440 rows x 8 columns]"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "logs = f\"{WS}/{LOGS}/QC\"\n",
    "summ = pd.DataFrame()\n",
    "summ.index.name='SampleID'\n",
    "for clean in cleans:\n",
    "    if clean in [\"\"]:\n",
    "        continue\n",
    "    file = [f for f in os.listdir(logs) if f.startswith(f\"{clean}_QC.sh.e\")][0]\n",
    "    with open(f'{logs}/{file}') as f:\n",
    "        txt = f.read().split('Total number of sequences analysed: ')[-1]\n",
    "\n",
    "        #Get QC stats\n",
    "        total = float(txt.split('\\n')[0])\n",
    "        bad = float(txt.split('bp): ')[-1].split(' (')[0])\n",
    "        \n",
    "        #Get Host/Feed DNA\n",
    "        no_host = float(txt.split('-\\n    ')[1].split(' ')[0])\n",
    "        no_feed = float(txt.split('-\\n    ')[3].split(' ')[0])\n",
    "        \n",
    "    summ.loc[clean, ['Total', 'PassedQC', 'NoHost', 'NoFeed']] = [total, total-bad, no_host, no_feed]\n",
    "\n",
    "#Count %\n",
    "summ['PassedQC(%)'] = round(100*summ.PassedQC/summ.Total, 3)\n",
    "summ['Host(%_of_PassedQC)'] = round(100*(summ.PassedQC-summ.NoHost)/summ.PassedQC, 3)\n",
    "summ['Feed(%_of_PassedQC)'] = round(100*(summ.NoHost-summ.NoFeed)/summ.PassedQC, 3)\n",
    "\n",
    "\n",
    "#add owner of samples\n",
    "names = pd.read_csv(f'{WS}/sampleName_clientId.txt', sep='\\t', index_col='clientId')\n",
    "summ['Owner'] = names['Owner']\n",
    "summ.to_csv(f'{WS}/{DATA}/QC_stats.tsv', sep='\\t')\n",
    "summ"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "87920830-d53b-4959-b09d-d17320bc8ec7",
   "metadata": {},
   "source": [
    "# Metagenome assemblies Megahit"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5f317391-6385-4e27-8491-786b80ef7731",
   "metadata": {},
   "outputs": [],
   "source": [
    "THREADS = 12\n",
    "RAM = 90\n",
    "logdir = f'{LOGS}/MegaHit'\n",
    "\n",
    "todo = '''\n",
    "conda activate qiime2-shotgun-2024.2\n",
    "echo \"Processing {BATCH}\"\n",
    "cd $WS\n",
    "mkdir -p $MH\n",
    "\n",
    "sras=({IDS})\n",
    "for name in ${{sras[@]}}; do\n",
    "    echo\n",
    "    echo \"############################################################\"\n",
    "    echo \"Processing $name \"\n",
    "    \n",
    "    echo \"Check if reads already assembled\" \n",
    "    if [ -f $ASS/${{name}}.fa ]; then\n",
    "        echo \"$name already assembled. Skipping.\"\n",
    "        continue\n",
    "    fi\n",
    "    echo \"$name is not assembled. Attempting assembly...\"\n",
    "    \n",
    "    echo \"Check if reads are paired or single\" \n",
    "    if [ -f $CR/${{name}}_1.fq.gz ]; then\n",
    "        echo \"$name are paired reads.\"\n",
    "        CR1=$CR/${{name}}_1.fq.gz\n",
    "        CR2=$CR/${{name}}_2.fq.gz\n",
    "    \n",
    "        megahit -1 $CR1 -2 $CR2 -o $MH/$name -t {THREADS} --presets meta-sensitive\n",
    "    fi\n",
    "    \n",
    "    if [ -f $CR/${{name}}.fq.gz ]; then\n",
    "        echo \"$name are single reads.\"\n",
    "        CR1=$CR/${{name}}.fq.gz\n",
    "    \n",
    "        megahit -r $CR1 -o $MH/$name -t {THREADS} --presets meta-sensitive\n",
    "    fi\n",
    "\n",
    "    echo \"Checking if assembly created\"\n",
    "    ass=$MH/$name/final.contigs.fa\n",
    "    if [ -f $ass ]; then\n",
    "        echo \"Moving assembly\"\n",
    "        mv $ass $ASS/${{name}}.fa\n",
    "    else\n",
    "        echo \"There is something wrong with assembly\"\n",
    "    fi\n",
    "    rm -r $MH/$name\n",
    "done\n",
    "\n",
    "echo\n",
    "echo \"############################################################\"\n",
    "echo \"###                   All done! Enjoy!                   ###\"\n",
    "echo \"### Citations:                                           ###\"\n",
    "echo \"### 1. MegaHit:                                          ###\"\n",
    "echo \"###    https://doi.org/10.1093/bioinformatics/btv033     ###\"\n",
    "echo \"############################################################\"   \n",
    "'''\n",
    "\n",
    "#launch samples\n",
    "for batch, ids in batches.items():\n",
    "    if batch in ['batch1-30']:\n",
    "        continue\n",
    "\n",
    "    IDS = ' '.join(ids)\n",
    "    binac(ppn=THREADS, time='07:00:00:00', mem=RAM, name=f'{batch}_MH', queue='long', \n",
    "          logdir=logdir, todo=todo.format(BATCH=batch, IDS=IDS, THREADS=THREADS))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "610c7a10-49fc-4ae6-8735-3fa1a8e30ba3",
   "metadata": {},
   "outputs": [],
   "source": [
    "#quast -t {THREADS} -o {out}/Quast {out}/final.contigs.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "d17275be-4eb3-40db-a1e5-651cfa083dc6",
   "metadata": {},
   "outputs": [],
   "source": [
    "for clean in cleans:\n",
    "    out = f'{WS}/Results/MegaHit/{clean}'\n",
    "    !rm -r $out/intermediate_contigs\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "7b7aa954-8379-45e5-abd9-3db48432ec9e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11242915\n"
     ]
    }
   ],
   "source": [
    "#zip assemblies\n",
    "\n",
    "logs = f\"{LOGS}/MegaHit\"\n",
    "\n",
    "todo = '''\n",
    "conda activate Work\n",
    "cd $WS\n",
    "\n",
    "zip -r MegaHit.zip Results/MegaHit\n",
    "\n",
    "'''\n",
    "\n",
    "binac(time='00:48:00:00', mem=16, name=f'MegaHit_zip', queue='short', logdir=logs,\n",
    "     todo=todo)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "edc99c0c-5d52-431f-a03b-4df738635d9e",
   "metadata": {},
   "source": [
    "# Status"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "4feb5bf8-ea1b-46b9-a060-cf4f1cca84cd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "mgmt02: \n",
      "                                                                                  Req'd       Req'd       Elap\n",
      "Job ID                  Username    Queue    Jobname          SessID  NDS   TSK   Memory      Time    S   Time\n",
      "----------------------- ----------- -------- ---------------- ------ ----- ------ --------- --------- - ---------\n",
      "11327685                ho_graaf20  smp      BLAST-taxa.sh     28410     1     24    1000gb 720:00:00 R 568:34:03\n",
      "11339514                ho_graaf20  smp      SM_MAGs.sh        39589     1     12     256gb 720:00:00 R 405:57:55\n",
      "11376500                ho_graaf20  long     batch541-570_MH.   8585     1     12      90gb 168:00:00 R 139:24:08\n",
      "11376501                ho_graaf20  long     batch571-600_MH.   8390     1     12      90gb 168:00:00 R 136:26:16\n",
      "11376502                ho_graaf20  long     batch601-630_MH.  31769     1     12      90gb 168:00:00 R 135:51:01\n",
      "11376503                ho_graaf20  long     batch631-660_MH.   1157     1     12      90gb 168:00:00 R 135:48:58\n",
      "11376505                ho_graaf20  long     batch691-720_MH.  12411     1     12      90gb 168:00:00 R 135:39:46\n",
      "11376506                ho_graaf20  long     batch721-750_MH.  19699     1     12      90gb 168:00:00 R 135:37:44\n",
      "11376507                ho_graaf20  long     batch751-780_MH.   8570     1     12      90gb 168:00:00 R 135:36:43\n",
      "11376530                ho_graaf20  long     batch1411-1440_M  15650     1     12      90gb 168:00:00 R 135:19:14\n"
     ]
    }
   ],
   "source": [
    "!qstat -u ho_graaf20"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "id": "8a9057a8-7267-4ce7-8f5b-e3052ee31a09",
   "metadata": {},
   "outputs": [],
   "source": [
    "!qdel 11366477"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 313,
   "id": "97f32da6-01df-4ff7-99e7-8f723b16e7e6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "usage: qdel [{ -a | -c | -p | -t | -W delay | -m message}] [-b retry_seconds] [<JOBID>[<JOBID>]|'all'|'ALL']...\n",
      "       -a -c, -m, -p, -t, and -W are mutually exclusive\n",
      "usage: qdel [{ -a | -c | -p | -t | -W delay | -m message}] [-b retry_seconds] [<JOBID>[<JOBID>]|'all'|'ALL']...\n",
      "       -a -c, -m, -p, -t, and -W are mutually exclusive\n"
     ]
    }
   ],
   "source": [
    "kill =  \"\"\"\n",
    "11376394                ho_graaf20  long     batch31-60_MH.sh    --      1     24     120gb 168:00:00 Q       -- \n",
    "11376395                ho_graaf20  long     batch61-90_MH.sh    --      1     24     120gb 168:00:00 Q       -- \n",
    "11376396                ho_graaf20  long     batch91-120_MH.s    --      1     24     120gb 168:00:00 Q       -- \n",
    "11376397                ho_graaf20  long     batch121-150_MH.    --      1     24     120gb 168:00:00 Q       --  \n",
    "\"\"\"\n",
    "\n",
    "for job in kill.split('\\n'):\n",
    "    ID = job.split(' ')[0]\n",
    "    !qdel $ID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "id": "a55d9a0f-6bc8-4987-bb5e-7e9557641e24",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "qstat: Unknown Job Id Error 11366127.mgmt\n"
     ]
    }
   ],
   "source": [
    "!qstat 11366127  \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "67da283e-b146-4859-8bb2-54e1676f00c4",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job ID                    Name             User            Time Use S Queue\n",
      "------------------------- ---------------- --------------- -------- - -----\n",
      "10161757[]                 fb_rna(1)        ho_braat76             0 C short          \n",
      "10899958                   h2681L14         hd_wu263               0 H long           \n",
      "11089184                   STDIN            tu_zxmle82             0 Q gpu            \n",
      "11151211                   Cluster_reads.sh ho_graaf20      3871:14: R smp            \n",
      "11153399                   adec_0           st_uws86178     2610:53: R long           \n",
      "11153400                   adec_1           st_uws86178     343:09:4 R long           \n",
      "11153405                   adec_0           st_uws86178     2607:08: R long           \n",
      "11153406                   adec_1           st_uws86178     341:05:3 R long           \n",
      "11154060                   ...-tuebingen.de tu_emins01      556:45:2 R gpu            \n",
      "11154619                   go_allLambda.sh  st_st176732     1710:25: R long           \n",
      "11154620                   go_allLambda.sh  st_st176732     1678:08: R long           \n",
      "11154621                   go_allLambda.sh  st_st176732     1715:29: R long           \n",
      "11154622                   go_allLambda.sh  st_st176732     1677:51: R long           \n",
      "11154623                   go_allLambda.sh  st_st176732     1667:07: R long           \n",
      "11154624                   go_allLambda.sh  st_st176732     1634:18: R long           \n",
      "11154625                   go_allLambda.sh  st_st176732     1612:37: R long           \n",
      "11154626                   go_allLambda.sh  st_st176732     1622:11: R long           \n",
      "11154627                   go_allLambda.sh  st_st176732     1086:42: R long           \n",
      "11154628                   go_allLambda.sh  st_st176732     1003:49: R long           \n",
      "11154629                   go_allLambda.sh  st_st176732     998:38:0 R long           \n",
      "11154630                   go_allLambda.sh  st_st176732     1031:48: R long           \n",
      "11154631                   go_allLambda.sh  st_st176732     1027:19: R long           \n",
      "11154632                   go_allLambda.sh  st_st176732     999:19:1 R long           \n",
      "11154633                   go_allLambda.sh  st_st176732     1002:33: R long           \n",
      "11154634                   go_allLambda.sh  st_st176732     1000:59: R long           \n",
      "11154635                   go_allLambda.sh  st_st176732     976:55:4 R long           \n",
      "11154636                   go_allLambda.sh  st_st176732     936:25:0 R long           \n",
      "11155238                   ...-Hera_longsml tu_pecsc01      70:43:18 R gpu            \n",
      "11155239                   WGSBAC_Test      ho_holzer       385:36:3 R gpu            \n",
      "11155243                   fidu-test        tu_pecsc01      69:33:13 R gpu            \n",
      "11155487                   T1S1C0B0_fA.sh   hd_bu271        21290:09 R long           \n",
      "11225339                   aoc_0            st_uws86178     955:50:2 R short          \n",
      "11225340                   aoc_1            st_uws86178     129:08:2 R short          \n",
      "11225341                   adec_0           st_uws86178     1058:31: R short          \n",
      "11225347                   at8tdp-pa        st_ac127777     43:18:06 R gpu            \n",
      "11228344                   pwects           tu_beesc01      280:56:1 R long           \n",
      "11228346                   pwects           tu_beesc01      250:39:2 R long           \n",
      "11228389                   pwects           tu_beesc01      251:36:4 R long           \n",
      "11228437                   npt_out          st_ac141462     524:21:0 R short          \n",
      "11228438                   pwects           tu_beesc01      240:03:4 R long           \n",
      "11228440                   npt_out          st_ac141462     522:50:0 R short          \n",
      "11228441                   npt_out          st_ac141462     523:00:3 R short          \n",
      "11228442                   npt_out          st_ac141462     522:52:5 R short          \n",
      "11228444                   npt_out          st_ac141462     522:26:4 R short          \n",
      "11228445                   ...ediate_python st_st163811     00:00:36 R long           \n",
      "11228450                   NS_k2_-0000      tu_pelol01      16620:37 R short          \n",
      "11228453                   velocyto         tu_bbcta01      05:38:51 R short          \n",
      "11228455                   pwects           tu_beesc01      228:55:2 R long           \n",
      "11228459                   pwects           tu_beesc01      58:56:09 R long           \n",
      "11228460                   pwects           tu_beesc01      230:53:4 R long           \n",
      "11228461                   pwects           tu_beesc01      219:12:2 R long           \n",
      "11228462                   pwects           tu_beesc01      219:37:3 R long           \n",
      "11228463                   velocyto         tu_bbcta01      05:07:24 R short          \n",
      "11228501                   gm_test          hd_kk263        208:23:2 R long           \n",
      "11228504                   velocyto         tu_bbcta01      05:04:34 R short          \n",
      "11228505                   velocyto         tu_bbcta01      05:20:11 R short          \n",
      "11228506                   velocyto         tu_bbcta01      05:06:42 R short          \n",
      "11228507                   velocyto         tu_bbcta01      06:02:25 R short          \n",
      "11228508                   butane_job.sh    st_uws86178     277:36:0 C short          \n",
      "11228509                   butane_job.sh    st_uws86178     276:18:0 C short          \n",
      "11228510                   butane_job.sh    st_uws86178     277:32:1 C short          \n",
      "11228517                   butane_job.sh    st_uws86178     276:08:0 C short          \n",
      "11228520                   butane_job.sh    st_uws86178     277:25:1 C short          \n",
      "11228548                   pentane_job.sh   st_uws86178     277:28:4 R short          \n",
      "11228549                   pentane_job.sh   st_uws86178     277:18:3 R short          \n",
      "11228550                   pentane_job.sh   st_uws86178     277:34:0 R short          \n",
      "11228551                   pentane_job.sh   st_uws86178     277:27:4 R short          \n",
      "11228552                   pentane_job.sh   st_uws86178     277:33:4 R short          \n",
      "11228553                   pentane_job.sh   st_uws86178     277:21:2 R short          \n",
      "11228554                   pentane_job.sh   st_uws86178     277:25:3 R short          \n",
      "11228555                   pentane_job.sh   st_uws86178     277:27:3 R short          \n",
      "11228556                   pentane_job.sh   st_uws86178     277:30:2 R short          \n",
      "11228557                   pentane_job.sh   st_uws86178     277:22:5 R short          \n",
      "11228558                   pentane_job.sh   st_uws86178     277:23:1 R short          \n",
      "11228559                   pentane_job.sh   st_uws86178     277:29:1 R short          \n",
      "11228560                   pentane_job.sh   st_uws86178     277:17:2 R short          \n",
      "11228561                   pentane_job.sh   st_uws86178     277:33:5 R short          \n",
      "11228562                   pentane_job.sh   st_uws86178     277:20:0 R short          \n",
      "11228563                   pentane_job.sh   st_uws86178     277:33:5 R short          \n",
      "11228564                   pentane_job.sh   st_uws86178     277:29:5 R short          \n",
      "11228565                   pentane_job.sh   st_uws86178     277:23:3 R short          \n",
      "11228566                   pentane_job.sh   st_uws86178     277:24:4 R short          \n",
      "11228567                   pentane_job.sh   st_uws86178     277:20:5 R short          \n",
      "11228568                   pentane_job.sh   st_uws86178     145:57:1 R short          \n",
      "11228569                   pentane_job.sh   st_uws86178     39:47:22 R short          \n",
      "11228570                   pentane_job.sh   st_uws86178     38:53:03 R short          \n",
      "11228571                   pentane_job.sh   st_uws86178     39:12:23 R short          \n",
      "11228572                   pentane_job.sh   st_uws86178     39:08:39 R short          \n",
      "11228573                   pentane_job.sh   st_uws86178     39:08:15 R short          \n",
      "11228574                   pentane_job.sh   st_uws86178     38:15:30 R short          \n",
      "11228575                   pentane_job.sh   st_uws86178     37:59:36 R short          \n",
      "11228576                   pentane_job.sh   st_uws86178     38:02:55 R short          \n",
      "11228577                   pentane_job.sh   st_uws86178     38:15:04 R short          \n",
      "11228578                   pentane_job.sh   st_uws86178     38:08:28 R short          \n",
      "11228579                   pentane_job.sh   st_uws86178     37:55:34 R short          \n",
      "11228580                   pentane_job.sh   st_uws86178     38:00:12 R short          \n",
      "11228581                   pentane_job.sh   st_uws86178     37:48:55 R short          \n",
      "11228582                   pentane_job.sh   st_uws86178     37:12:56 R short          \n",
      "11228583                   pentane_job.sh   st_uws86178     37:15:20 R short          \n",
      "11228584                   pentane_job.sh   st_uws86178     36:59:25 R short          \n",
      "11228585                   pentane_job.sh   st_uws86178     36:30:39 R short          \n",
      "11228586                   pentane_job.sh   st_uws86178     36:38:31 R short          \n",
      "11228587                   pentane_job.sh   st_uws86178     36:13:18 R short          \n",
      "11228588                   pentane_job.sh   st_uws86178     36:12:01 R short          \n",
      "11228589                   pentane_job.sh   st_uws86178     04:39:29 R short          \n",
      "11228590                   pentane_job.sh   st_uws86178     04:26:27 R short          \n",
      "11228591                   pentane_job.sh   st_uws86178     03:40:11 R short          \n",
      "11228592                   pentane_job.sh   st_uws86178     03:32:19 R short          \n",
      "11228593                   pentane_job.sh   st_uws86178     03:52:26 R short          \n",
      "11228594                   pentane_job.sh   st_uws86178     03:50:05 R short          \n",
      "11228595                   pentane_job.sh   st_uws86178     03:08:31 R short          \n",
      "11228596                   pentane_job.sh   st_uws86178     02:52:57 R short          \n",
      "11228597                   pentane_job.sh   st_uws86178     02:58:06 R short          \n",
      "11228598                   octane_job.sh    st_uws86178     02:16:03 R short          \n",
      "11228599                   octane_job.sh    st_uws86178     02:38:57 R short          \n",
      "11228600                   octane_job.sh    st_uws86178     02:34:13 R short          \n",
      "11228601                   octane_job.sh    st_uws86178     02:19:18 R short          \n",
      "11228602                   octane_job.sh    st_uws86178     02:00:09 R short          \n",
      "11228603                   octane_job.sh    st_uws86178     01:20:25 R short          \n",
      "11228604                   octane_job.sh    st_uws86178     01:35:50 R short          \n",
      "11228605                   octane_job.sh    st_uws86178     00:56:31 R short          \n",
      "11228606                   octane_job.sh    st_uws86178     00:11:44 R short          \n",
      "11228607                   octane_job.sh    st_uws86178     00:04:17 R short          \n",
      "11228608                   octane_job.sh    st_uws86178            0 R short          \n",
      "11228609                   octane_job.sh    st_uws86178            0 Q short          \n",
      "11228610                   octane_job.sh    st_uws86178            0 Q short          \n",
      "11228611                   octane_job.sh    st_uws86178            0 Q short          \n",
      "11228612                   octane_job.sh    st_uws86178            0 Q short          \n",
      "11228613                   octane_job.sh    st_uws86178            0 Q short          \n",
      "11228614                   octane_job.sh    st_uws86178            0 Q short          \n",
      "11228615                   octane_job.sh    st_uws86178            0 Q short          \n",
      "11228616                   octane_job.sh    st_uws86178            0 Q short          \n",
      "11228617                   octane_job.sh    st_uws86178            0 Q short          \n",
      "11228618                   octane_job.sh    st_uws86178            0 Q short          \n",
      "11228619                   octane_job.sh    st_uws86178            0 Q short          \n",
      "11228620                   octane_job.sh    st_uws86178            0 Q short          \n",
      "11228621                   octane_job.sh    st_uws86178            0 Q short          \n",
      "11228622                   octane_job.sh    st_uws86178            0 Q short          \n",
      "11228623                   octane_job.sh    st_uws86178            0 Q short          \n",
      "11228624                   octane_job.sh    st_uws86178            0 Q short          \n",
      "11228625                   octane_job.sh    st_uws86178            0 Q short          \n",
      "11228626                   octane_job.sh    st_uws86178            0 Q short          \n",
      "11228627                   octane_job.sh    st_uws86178            0 Q short          \n",
      "11228628                   octane_job.sh    st_uws86178            0 Q short          \n",
      "11228629                   octane_job.sh    st_uws86178            0 Q short          \n",
      "11228630                   octane_job.sh    st_uws86178            0 Q short          \n",
      "11228631                   octane_job.sh    st_uws86178            0 Q short          \n",
      "11228632                   octane_job.sh    st_uws86178            0 Q short          \n",
      "11228633                   octane_job.sh    st_uws86178            0 Q short          \n",
      "11228634                   octane_job.sh    st_uws86178            0 Q short          \n",
      "11228635                   octane_job.sh    st_uws86178            0 Q short          \n",
      "11228636                   octane_job.sh    st_uws86178            0 Q short          \n",
      "11228637                   octane_job.sh    st_uws86178            0 Q short          \n",
      "11228638                   octane_job.sh    st_uws86178            0 Q short          \n",
      "11228639                   octane_job.sh    st_uws86178            0 Q short          \n",
      "11228640                   octane_job.sh    st_uws86178            0 Q short          \n",
      "11228641                   octane_job.sh    st_uws86178            0 Q short          \n",
      "11228642                   octane_job.sh    st_uws86178            0 Q short          \n",
      "11228643                   octane_job.sh    st_uws86178            0 Q short          \n",
      "11228644                   octane_job.sh    st_uws86178            0 Q short          \n",
      "11228645                   octane_job.sh    st_uws86178            0 Q short          \n",
      "11228646                   octane_job.sh    st_uws86178            0 Q short          \n",
      "11228647                   octane_job.sh    st_uws86178            0 Q short          \n",
      "11228648                   hexane_job.sh    st_uws86178            0 Q short          \n",
      "11228649                   hexane_job.sh    st_uws86178            0 Q short          \n",
      "11228650                   hexane_job.sh    st_uws86178            0 Q short          \n",
      "11228651                   hexane_job.sh    st_uws86178            0 Q short          \n",
      "11228652                   hexane_job.sh    st_uws86178            0 Q short          \n",
      "11228653                   hexane_job.sh    st_uws86178            0 Q short          \n",
      "11228654                   hexane_job.sh    st_uws86178            0 Q short          \n",
      "11228655                   hexane_job.sh    st_uws86178            0 Q short          \n",
      "11228656                   hexane_job.sh    st_uws86178            0 Q short          \n",
      "11228657                   hexane_job.sh    st_uws86178            0 Q short          \n",
      "11228658                   hexane_job.sh    st_uws86178            0 Q short          \n",
      "11228659                   hexane_job.sh    st_uws86178            0 Q short          \n",
      "11228660                   hexane_job.sh    st_uws86178            0 Q short          \n",
      "11228661                   hexane_job.sh    st_uws86178            0 Q short          \n",
      "11228662                   hexane_job.sh    st_uws86178            0 Q short          \n",
      "11228663                   hexane_job.sh    st_uws86178            0 Q short          \n",
      "11228664                   hexane_job.sh    st_uws86178            0 Q short          \n",
      "11228665                   hexane_job.sh    st_uws86178            0 Q short          \n",
      "11228666                   hexane_job.sh    st_uws86178            0 Q short          \n",
      "11228667                   hexane_job.sh    st_uws86178            0 Q short          \n",
      "11228668                   hexane_job.sh    st_uws86178            0 Q short          \n",
      "11228669                   hexane_job.sh    st_uws86178            0 Q short          \n",
      "11228670                   hexane_job.sh    st_uws86178            0 Q short          \n",
      "11228671                   hexane_job.sh    st_uws86178            0 Q short          \n",
      "11228672                   hexane_job.sh    st_uws86178            0 Q short          \n",
      "11228673                   hexane_job.sh    st_uws86178            0 Q short          \n",
      "11228674                   hexane_job.sh    st_uws86178            0 Q short          \n",
      "11228675                   hexane_job.sh    st_uws86178            0 Q short          \n",
      "11228676                   hexane_job.sh    st_uws86178            0 Q short          \n",
      "11228677                   hexane_job.sh    st_uws86178            0 Q short          \n",
      "11228678                   hexane_job.sh    st_uws86178            0 Q short          \n",
      "11228679                   hexane_job.sh    st_uws86178            0 Q short          \n",
      "11228680                   hexane_job.sh    st_uws86178            0 Q short          \n",
      "11228681                   hexane_job.sh    st_uws86178            0 Q short          \n",
      "11228682                   hexane_job.sh    st_uws86178            0 Q short          \n",
      "11228683                   hexane_job.sh    st_uws86178            0 Q short          \n",
      "11228684                   hexane_job.sh    st_uws86178            0 Q short          \n",
      "11228685                   hexane_job.sh    st_uws86178            0 Q short          \n",
      "11228686                   hexane_job.sh    st_uws86178            0 Q short          \n",
      "11228687                   hexane_job.sh    st_uws86178            0 Q short          \n",
      "11228688                   hexane_job.sh    st_uws86178            0 Q short          \n",
      "11228689                   hexane_job.sh    st_uws86178            0 Q short          \n",
      "11228690                   hexane_job.sh    st_uws86178            0 Q short          \n",
      "11228691                   hexane_job.sh    st_uws86178            0 Q short          \n",
      "11228692                   hexane_job.sh    st_uws86178            0 Q short          \n",
      "11228693                   hexane_job.sh    st_uws86178            0 Q short          \n",
      "11228694                   hexane_job.sh    st_uws86178            0 Q short          \n",
      "11228695                   hexane_job.sh    st_uws86178            0 Q short          \n",
      "11228696                   hexane_job.sh    st_uws86178            0 Q short          \n",
      "11228697                   hexane_job.sh    st_uws86178            0 Q short          \n",
      "11228701                   sim_vdw_1        st_st163811     84:24:11 R short          \n",
      "11228702                   sim_vdw_3        st_st163811     84:27:00 R short          \n",
      "11228703                   sim_vdw_5        st_st163811     84:27:20 R short          \n",
      "11228782                   baymodts         st_ac138559     01:52:14 R long           \n",
      "11228783                   high_reso        tu_pecaf01      21:06:35 R long           \n",
      "11228784                   q2_taxonomy.sh   ho_graaf20             0 Q smp            \n"
     ]
    }
   ],
   "source": [
    "!qstat"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
