{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "f57487f7-3f08-4d3d-92a2-e45daa08ed4b",
   "metadata": {},
   "source": [
    "# Variables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "004b661e-b73f-47a3-86ea-a8b5c3420a89",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of raw samples:  0\n",
      "Number of clean samples:  242\n"
     ]
    }
   ],
   "source": [
    "### Managing work ###\n",
    "\n",
    "NAME = 'ChickenPhages'\n",
    "\n",
    "#!ws_allocate {NAME} 30\n",
    "#!ws_list -a\n",
    "#!ws_find {NAME}\n",
    "#!ws_extend {NAME}\n",
    "#!ws_release {NAME} #delete, remove content first\n",
    "\n",
    "\n",
    "VARS = (\n",
    "    f'WS=/pfs/10/work/ho_graaf20-{NAME}/\\n'\n",
    "    f'PS=/pfs/10/project/bw18d010/ho_graaf20/{NAME}/\\n'\n",
    "    f'DB=/pfs/10/project/bw18d010/ho_graaf20/Databases/\\n'\n",
    "    f'REFS=/pfs/10/project/bw18d010/ho_graaf20/Databases/Reference_Genomes\\n'\n",
    "    f'LOGS=/pfs/10/project/bw18d010/ho_graaf20/{NAME}/Logs\\n'\n",
    "    f'RR=/pfs/10/project/bw18d010/ho_graaf20/{NAME}/Raw_reads\\n'\n",
    "    f'CR=/pfs/10/project/bw18d010/ho_graaf20/{NAME}/Clean_reads\\n'\n",
    ")\n",
    "\n",
    "CONDA = '$HOME/miniforge3/etc/profile.d/conda.sh'\n",
    "\n",
    "###############################################################################\n",
    "################################ DO NOT CHANGE ################################\n",
    "###############################################################################\n",
    "\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import os\n",
    "\n",
    "#extract bash variables for python\n",
    "for var in VARS.strip().split('\\n'):\n",
    "    vars()[var.split('=')[0]]=var.split('=')[1]\n",
    "\n",
    "!mkdir -p {CR} {LOGS} {RR} {DB}\n",
    "!ln -sf -T {PS} ./'PS'\n",
    "!ln -sf -T {WS} ./'WS'\n",
    "!ln -sf -T {DB} ~/'DB'\n",
    "\n",
    "    \n",
    "#Binac function\n",
    "source = f'#!/bin/bash\\nsource {CONDA}'\n",
    "COMM = 'export TMPDIR=$TMPDIR\\n#Custom variables and commands\\ncd $WS\\n'\n",
    "STRING = '\\n'.join([source, VARS, COMM])\n",
    "\n",
    "# binac function that will submit the job to server\n",
    "def binac2(tasks=1, ppn=1, time='00:15:00', mem=2, name='myjob', logdir=LOGS,\n",
    "          queue='compute', todo='echo \"Command\"', STRING=STRING, nodes=1):\n",
    "    text = '\\n'.join([STRING, todo])\n",
    "    script = f'{logdir}/{name}.sh'\n",
    "    !mkdir -p {logdir}\n",
    "    with open(script, \"w\") as text_file:\n",
    "        text_file.write(text)\n",
    "    !chmod +x {script}\n",
    "    run = f\"\"\"sbatch -t {time} -p {queue} --mem={mem}g -J {name} --output={logdir}/{name}.log \\\n",
    "              --error={logdir}/{name}.error --ntasks-per-node={tasks} -N {nodes} {script}\"\"\"\n",
    "    !$run\n",
    "\n",
    "cleaning = !rm -r $WS/*/*/.ipynb_c* $WS/*/.ipynb_c* $WS/.ipynb_c* $PS/*/*/.ipynb_c* $PS/*/.ipynb_c* $PS/.ipynb_c*\n",
    "\n",
    "###############################################################################\n",
    "######################## Here you can change something ########################\n",
    "###############################################################################\n",
    "\n",
    "meta = pd.read_csv(\"PS/MAGs_dataset_hosts.tsv\", sep='\\t', index_col=0)\n",
    "hosts = meta.isolation_host.unique().tolist()\n",
    "sras = meta.index.tolist()\n",
    "b = 50\n",
    "batches = {f'batch{i+1}-{i+b}': sras[i:i + b] for i in range(0, len(sras), b)}\n",
    "\n",
    "raws = []\n",
    "if os.path.exists(RR): \n",
    "    raws = [f.rsplit('_', 1)[0] for f in os.listdir(RR) if '.gz' in f]\n",
    "    raws = sorted(list(set(raws)))\n",
    "    print('Number of raw samples: ', len(raws))\n",
    "\n",
    "cleans = []\n",
    "if os.path.exists(CR): \n",
    "    cleans = [f.rsplit('_', 1)[0] for f in os.listdir(CR) if '.gz' in f]\n",
    "    cleans = sorted(list(set(cleans)))\n",
    "    print('Number of clean samples: ', len(cleans))\n",
    "\n",
    "bad = []"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "31773337-0dbf-4abe-a75f-a5d330d0247e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "242"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(meta)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "440f0918-0c47-484b-b6d7-aebd6aa9a756",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cow 37\n",
      "duck 42\n",
      "human 45\n",
      "pig 48\n",
      "turkey 70\n"
     ]
    }
   ],
   "source": [
    "for host in hosts:\n",
    "    print(host, len(meta.loc[meta.isolation_host==host]))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ab31df15-c3e8-46e1-bb2a-a0c91a9fd293",
   "metadata": {},
   "source": [
    "# Download samples by Hosts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cb241b3c-9e14-406d-909d-442f93730d09",
   "metadata": {},
   "outputs": [],
   "source": [
    "#conda create -n SRA -c bioconda sra-tools pigz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "4ee401a6-2bfc-4cae-9938-0e76436f87cb",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Submitted batch job 38549\n",
      "Submitted batch job 38550\n",
      "Submitted batch job 38551\n",
      "Submitted batch job 38552\n"
     ]
    }
   ],
   "source": [
    "#use only 1 thread or bad things may happen (Binac will freeze)!\n",
    "\n",
    "THREADS = 2\n",
    "logdir = f'{LOGS}/SRAs'\n",
    "\n",
    "todo = '''\n",
    "conda activate SRA\n",
    "\n",
    "mkdir -p sra_temp\n",
    "\n",
    "echo \"Processing {BATCH}\"\n",
    "sras=({IDS})\n",
    "for sra in ${{sras[@]}}; do\n",
    "\n",
    "    #skip processed files\n",
    "    if [ -f $RR/${{sra}}_1.fastq.gz ]; then\n",
    "        echo \"$sra already exists. Skipping.\"\n",
    "        continue\n",
    "    fi\n",
    "    \n",
    "    if [ -f $RR/${{sra}}.fastq.gz ]; then\n",
    "        echo \"$sra already exists. Skipping.\"\n",
    "        continue\n",
    "    fi\n",
    "    \n",
    "    if [ -f $CR/${{sra}}_1.fq.gz ]; then\n",
    "        echo \"$sra already exists. Skipping.\"\n",
    "        continue\n",
    "    fi\n",
    "    \n",
    "    if [ -f $CR/${{sra}}.fq.gz ]; then\n",
    "        echo \"$sra already exists. Skipping.\"\n",
    "        continue\n",
    "    fi\n",
    "    \n",
    "    #download files\n",
    "    echo \"Prefetch $sra\"\n",
    "    prefetch $sra -O sra_temp --verify yes --max-size 50G\n",
    "\n",
    "    echo \"Convert to fastq $sra\"\n",
    "    fasterq-dump sra_temp/$sra --split-files --outdir $RR --threads 1\n",
    "    rm -r sra_temp/$sra\n",
    "\n",
    "    echo \"Gzip $sra\"\n",
    "    pigz $RR/${{sra}}_*.fastq\n",
    "    pigz $RR/${{sra}}.fastq\n",
    "\n",
    "    \n",
    "done\n",
    "echo \"All done, check the outputs\"\n",
    "'''\n",
    "\n",
    "for batch, ids in batches.items():\n",
    "    if batch in ['batch1-50']:\n",
    "        continue\n",
    "        \n",
    "    IDS = ' '.join([i for i in ids if i not in bad])\n",
    "    binac2(tasks=THREADS, time='90:00:00', mem=8, name=f'{batch}_SRA', logdir=logdir, \n",
    "           todo=todo.format(BATCH=batch, IDS=IDS))\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "468a85b7-bc3c-4241-8951-d4825a8acce0",
   "metadata": {},
   "source": [
    "# Remove host DNA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ef9a6bd5-5987-41ee-aaf6-9c01d51bd8e6",
   "metadata": {},
   "outputs": [],
   "source": [
    "#conda create -n QC -c bioconda -c conda-forge trim-galore bowtie2 -y"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "b2a7970f-58e0-4546-8319-ba462d4b09cb",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Submitted batch job 38620\n",
      "Submitted batch job 38621\n",
      "Submitted batch job 38622\n",
      "Submitted batch job 38623\n",
      "Submitted batch job 38624\n",
      "Submitted batch job 38625\n"
     ]
    }
   ],
   "source": [
    "#create index\n",
    "\n",
    "THREADS = 8\n",
    "logdir = f'{LOGS}/Bowtie2'\n",
    "\n",
    "todo = '''\n",
    "conda activate QC\n",
    "cd $REFS/{index}\n",
    "\n",
    "#skip if index exists\n",
    "if [ -f $REFS/{index}/*.bt2 ]\n",
    "then\n",
    "    echo \"Index {index} already exists.\"\n",
    "else\n",
    "    pwd\n",
    "    echo \"Creating index {index}...\"\n",
    "    bowtie2-build ./*.fna.gz {index} --threads {THREADS}\n",
    "fi\n",
    "'''\n",
    "\n",
    "REF_GENS = ['Chicken', 'Corn', 'Cow', 'Duck', 'Human', 'Pig', 'Turkey']\n",
    "\n",
    "for index in REF_GENS:\n",
    "    if index == REF_GENS[0]:\n",
    "        continue\n",
    "    binac2(tasks=THREADS, time='10:00:00', mem=16, name=f'{index}_BW2_ind', logdir=logdir, \n",
    "           todo=todo.format(index=index, THREADS=THREADS))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "id": "26344e64-d887-49b3-96c0-a5de4b82b412",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Submitted batch job 38719\n",
      "Submitted batch job 38720\n",
      "Submitted batch job 38721\n",
      "Submitted batch job 38722\n",
      "Submitted batch job 38723\n"
     ]
    }
   ],
   "source": [
    "#Read QC and host DNA removal\n",
    "\n",
    "THREADS = 8\n",
    "logdir = f\"{LOGS}/QC\"\n",
    "\n",
    "todo = '''\n",
    "conda activate QC\n",
    "\n",
    "echo \"Processing {BATCH}\"\n",
    "sras=({IDS})\n",
    "for name in ${{sras[@]}}; do\n",
    "    ind_host={ind_host}\n",
    "    ind_feed={ind_feed}\n",
    "    preQC=$PS/FastQC/preQC/$name\n",
    "    postQC=$PS/FastQC/postQC/$name\n",
    "    outdir=$CR/$name\n",
    "    \n",
    "    echo\n",
    "    echo \"############################################################\"\n",
    "    echo \"Processing $name \"\n",
    "    echo \"Check if reads are paired or single\" \n",
    "\n",
    "    #skip processed files\n",
    "    if [ -d $outdir ]; then\n",
    "        echo \"$name already busy. Skipping.\"\n",
    "        continue\n",
    "    fi\n",
    "\n",
    "    cd $WS\n",
    "    if [ -f $RR/${{name}}_1.f*q.gz ]; then\n",
    "        echo \"$name are paired reads.\"\n",
    "        RR1=$RR/${{name}}_1.f*q.gz\n",
    "        RR2=$RR/${{name}}_2.f*q.gz\n",
    "        CR1=$CR/${{name}}_1.fq.gz\n",
    "        CR2=$CR/${{name}}_2.fq.gz\n",
    "    \n",
    "        #skip processed files\n",
    "        if [ -f $CR1 ]; then\n",
    "            echo \"$name already cleaned. Skipping.\"\n",
    "            continue\n",
    "        fi\n",
    "        \n",
    "        rm -rf $outdir $preQC $postQC $CR1 $CR2\n",
    "        mkdir -p $preQC $postQC $outdir \n",
    "    \n",
    "        # Fastq report before QC\n",
    "        echo \"Report before QC...\"\n",
    "        fastqc -q -t {THREADS} -o $preQC -f fastq $RR1 $RR2\n",
    "        \n",
    "        echo\n",
    "        echo \"Running Trim-Galore for QC\"\n",
    "        echo   \n",
    "        cd $outdir\n",
    "        trim_galore --paired $RR1 $RR2 -j {THREADS}\n",
    "    \n",
    "        # Fastq report after QC\n",
    "        echo \"Report after QC...\"\n",
    "        cd\n",
    "        cd $WS\n",
    "        fastqc -q -t {THREADS} -o $postQC -f fastq $outdir/*_val_1.f*q.gz $outdir/*_val_2.f*q.gz\n",
    "        \n",
    "        echo\n",
    "        echo \"Removing Host and Feed DNA from reads\"\n",
    "        echo \"Mapping reads to the host reference...\"\n",
    "        cd\n",
    "        cd $REFS/$ind_host\n",
    "        bowtie2 -p {THREADS} -x $ind_host -1 $outdir/*_val_1.f*q.gz -2 $outdir/*_val_2.f*q.gz \\\n",
    "        --un-conc-gz $outdir/no_host > $outdir/host.sam\n",
    "        echo \"Host DNA removed\"\n",
    "        \n",
    "        echo \"Mapping reads to the feed reference...\"\n",
    "        cd\n",
    "        cd $REFS/$ind_feed\n",
    "        bowtie2 -p {THREADS} -x $ind_feed -1 $outdir/no_host.1 -2 $outdir/no_host.2 \\\n",
    "        --un-conc-gz $outdir/no_feed > $outdir/feed.sam\n",
    "        echo \"Feed DNA removed\"\n",
    "        \n",
    "        echo\n",
    "        echo \"Move and clean!\"\n",
    "        cd\n",
    "        cd $WS\n",
    "        mv $outdir/no_feed.1 $CR1\n",
    "        mv $outdir/no_feed.2 $CR2\n",
    "        rm -rf $outdir\n",
    "    \n",
    "        #remove processed files\n",
    "        if [ -f $CR1 ]; then\n",
    "            echo \"Removing raw read since $name successfully cleaned.\"\n",
    "            rm $RR1 $RR2\n",
    "        fi\n",
    "    elif [ -f $RR/${{name}}.f*q.gz ]; then\n",
    "        echo \"$name are single reads.\"\n",
    "        RR1=$RR/${{name}}.f*q.gz\n",
    "        CR1=$CR/${{name}}.fq.gz\n",
    "    \n",
    "        #skip processed files\n",
    "        if [ -f $CR1 ]; then\n",
    "            echo \"$name already cleaned. Skipping.\"\n",
    "            continue\n",
    "        fi\n",
    "        \n",
    "        rm -rf $outdir $preQC $postQC $CR1 $CR2\n",
    "        mkdir -p $preQC $postQC $outdir \n",
    "    \n",
    "        # Fastq report before QC\n",
    "        echo \"Report before QC...\"\n",
    "        fastqc -q -t {THREADS} -o $preQC -f fastq $RR1\n",
    "        \n",
    "        echo\n",
    "        echo \"Running Trim-Galore for QC\"\n",
    "        echo   \n",
    "        cd $outdir\n",
    "        trim_galore $RR1 -j {THREADS}\n",
    "    \n",
    "        # Fastq report after QC\n",
    "        echo \"Report after QC...\"\n",
    "        cd\n",
    "        cd $WS\n",
    "        fastqc -q -t {THREADS} -o $postQC -f fastq $outdir/*_trimmed.f*q.gz\n",
    "        \n",
    "        echo\n",
    "        echo \"Removing Host and Feed DNA from reads\"\n",
    "        echo \"Mapping reads to the host reference...\"\n",
    "        cd\n",
    "        cd $REFS/$ind_host\n",
    "        bowtie2 -p {THREADS} -x $ind_host -U $outdir/*_trimmed.f*q.gz \\\n",
    "        --un-gz $outdir/no_host > $outdir/host.sam\n",
    "        echo \"Host DNA removed\"\n",
    "        \n",
    "        echo \"Mapping reads to the feed reference...\"\n",
    "        cd\n",
    "        cd $REFS/$ind_feed\n",
    "        bowtie2 -p {THREADS} -x $ind_feed -U $outdir/no_host \\\n",
    "        --un-gz $outdir/no_feed > $outdir/feed.sam\n",
    "        echo \"Feed DNA removed\"\n",
    "        \n",
    "        echo\n",
    "        echo \"Move and clean!\"\n",
    "        cd\n",
    "        cd $WS\n",
    "        mv $outdir/no_feed $CR1\n",
    "        rm -rf $outdir\n",
    "    \n",
    "        #remove processed files\n",
    "        if [ -f $CR1 ]; then\n",
    "            echo \"Removing raw read since $name successfully cleaned.\"\n",
    "            rm $RR1\n",
    "        fi\n",
    "        \n",
    "    else\n",
    "        echo \"There is something wrong with that sample (neither paired reads or single read were found)\"\n",
    "        continue\n",
    "        \n",
    "    fi\n",
    "done\n",
    "\n",
    "echo\n",
    "echo \"############################################################\"\n",
    "echo \"###                   All done! Enjoy!                   ###\"\n",
    "echo \"### Citations:                                           ###\"\n",
    "echo \"### 1. Cutadapt: https://doi.org/10.14806/ej.17.1.200    ###\"\n",
    "echo \"### 2. Bowtie: https://doi.org/10.1038/nmeth.1923        ###\"\n",
    "echo \"### 3. FastQC (optional):                                ###\"\n",
    "echo \"###    https://doi.org/10.1038/nmeth.1923                ###\"\n",
    "echo \"### 4. TrimGalore (optional):                            ###\"\n",
    "echo \"###    https://github.com/FelixKrueger/TrimGalore        ###\"\n",
    "echo \"############################################################\"\n",
    "'''\n",
    "\n",
    "\n",
    "#launch samples\n",
    "for host in hosts:\n",
    "    #if host not in ['turkey']:\n",
    "    #    continue\n",
    "        \n",
    "    IDS = ' '.join([i for i in meta.loc[meta.isolation_host == host].index if i in raws])\n",
    "            \n",
    "    binac2(tasks=THREADS, time='100:00:00', mem=32, name=f'{host}_QC8', logdir=logdir, \n",
    "           todo=todo.format(BATCH=host, IDS=IDS, ind_host=host.capitalize(), ind_feed='Corn', THREADS=THREADS))\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "de62600b-a4db-41e6-a7fd-1bff62624ec2",
   "metadata": {},
   "source": [
    "# CoverM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "45ed1ed4-24f0-4cb8-9ff6-4353efa8cb01",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Channels:\n",
      " - bioconda\n",
      " - conda-forge\n",
      "Platform: linux-64\n",
      "doneecting package metadata (repodata.json): - \n",
      "doneing environment: / \n",
      "\n",
      "## Package Plan ##\n",
      "\n",
      "  environment location: /home/ho/ho_genetics/ho_graaf20/miniforge3\n",
      "\n",
      "  added / updated specs:\n",
      "    - coverm\n",
      "\n",
      "\n",
      "The following packages will be downloaded:\n",
      "\n",
      "    package                    |            build\n",
      "    ---------------------------|-----------------\n",
      "    bwa-0.7.18                 |       h577a1d6_2         193 KB  bioconda\n",
      "    coverm-0.7.0               |       hb4818e0_2         2.2 MB  bioconda\n",
      "    dashing-1.0                |       h5b0a936_3         2.9 MB  bioconda\n",
      "    fastani-1.34               |       hb66fcc3_5         133 KB  bioconda\n",
      "    gsl-2.7                    |       he838d99_0         3.2 MB  conda-forge\n",
      "    htslib-1.21                |       h566b1c6_1         3.0 MB  bioconda\n",
      "    k8-1.2                     |       h6618dc6_3         7.2 MB  bioconda\n",
      "    kernel-headers_linux-64-3.10.0|      he073ed8_18         921 KB  conda-forge\n",
      "    minimap2-2.28              |       h577a1d6_4         1.2 MB  bioconda\n",
      "    openblas-0.3.28            |pthreads_h6ec200e_1         5.5 MB  conda-forge\n",
      "    samtools-1.21              |       h96c455f_1         476 KB  bioconda\n",
      "    starcode-1.4               |       h7b50bb2_6          39 KB  bioconda\n",
      "    strobealign-0.15.0         |       h5ca1c30_1         280 KB  bioconda\n",
      "    sysroot_linux-64-2.17      |      h0157908_18        14.5 MB  conda-forge\n",
      "    ------------------------------------------------------------\n",
      "                                           Total:        41.6 MB\n",
      "\n",
      "The following NEW packages will be INSTALLED:\n",
      "\n",
      "  bwa                bioconda/linux-64::bwa-0.7.18-h577a1d6_2 \n",
      "  coverm             bioconda/linux-64::coverm-0.7.0-hb4818e0_2 \n",
      "  dashing            bioconda/linux-64::dashing-1.0-h5b0a936_3 \n",
      "  fastani            bioconda/linux-64::fastani-1.34-hb66fcc3_5 \n",
      "  gsl                conda-forge/linux-64::gsl-2.7-he838d99_0 \n",
      "  htslib             bioconda/linux-64::htslib-1.21-h566b1c6_1 \n",
      "  isa-l              conda-forge/linux-64::isa-l-2.31.1-hb9d3cd8_1 \n",
      "  k8                 bioconda/linux-64::k8-1.2-h6618dc6_3 \n",
      "  kernel-headers_li~ conda-forge/noarch::kernel-headers_linux-64-3.10.0-he073ed8_18 \n",
      "  libdeflate         conda-forge/linux-64::libdeflate-1.23-h4ddbbb0_0 \n",
      "  minimap2           bioconda/linux-64::minimap2-2.28-h577a1d6_4 \n",
      "  openblas           conda-forge/linux-64::openblas-0.3.28-pthreads_h6ec200e_1 \n",
      "  perl               conda-forge/linux-64::perl-5.32.1-7_hd590300_perl5 \n",
      "  samtools           bioconda/linux-64::samtools-1.21-h96c455f_1 \n",
      "  starcode           bioconda/linux-64::starcode-1.4-h7b50bb2_6 \n",
      "  strobealign        bioconda/linux-64::strobealign-0.15.0-h5ca1c30_1 \n",
      "  sysroot_linux-64   conda-forge/noarch::sysroot_linux-64-2.17-h0157908_18 \n",
      "  zlib               conda-forge/linux-64::zlib-1.3.1-hb9d3cd8_2 \n",
      "\n",
      "\n",
      "\n",
      "Downloading and Extracting Packages:\n",
      "sysroot_linux-64-2.1 | 14.5 MB   |                                       |   0% \n",
      "k8-1.2               | 7.2 MB    |                                       |   0% \u001b[A\n",
      "\n",
      "openblas-0.3.28      | 5.5 MB    |                                       |   0% \u001b[A\u001b[A\n",
      "\n",
      "\n",
      "gsl-2.7              | 3.2 MB    |                                       |   0% \u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "htslib-1.21          | 3.0 MB    |                                       |   0% \u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "dashing-1.0          | 2.9 MB    |                                       |   0% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "coverm-0.7.0         | 2.2 MB    |                                       |   0% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "minimap2-2.28        | 1.2 MB    |                                       |   0% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "kernel-headers_linux | 921 KB    |                                       |   0% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "samtools-1.21        | 476 KB    |                                       |   0% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "strobealign-0.15.0   | 280 KB    |                                       |   0% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "bwa-0.7.18           | 193 KB    |                                       |   0% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "fastani-1.34         | 133 KB    |                                       |   0% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "sysroot_linux-64-2.1 | 14.5 MB   | 9                                     |   3% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "k8-1.2               | 7.2 MB    | #3                                    |   4% \u001b[A\n",
      "\n",
      "\n",
      "gsl-2.7              | 3.2 MB    | ##1                                   |   6% \u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "htslib-1.21          | 3.0 MB    | #####1                                |  14% \u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "openblas-0.3.28      | 5.5 MB    | 2                                     |   1% \u001b[A\u001b[A\n",
      "sysroot_linux-64-2.1 | 14.5 MB   | ##########4                           |  28% \u001b[A\n",
      "\n",
      "\n",
      "gsl-2.7              | 3.2 MB    | ###################################9  |  97% \u001b[A\u001b[A\u001b[A\n",
      "\n",
      "openblas-0.3.28      | 5.5 MB    | #############6                        |  37% \u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "htslib-1.21          | 3.0 MB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "htslib-1.21          | 3.0 MB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "gsl-2.7              | 3.2 MB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "sysroot_linux-64-2.1 | 14.5 MB   | ########################1             |  65% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "coverm-0.7.0         | 2.2 MB    | 2                                     |   1% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "k8-1.2               | 7.2 MB    | ##################################### | 100% \u001b[A\n",
      "k8-1.2               | 7.2 MB    | ##################################### | 100% \u001b[A\n",
      "\n",
      "openblas-0.3.28      | 5.5 MB    | ##################################### | 100% \u001b[A\u001b[A\n",
      "\n",
      "openblas-0.3.28      | 5.5 MB    | ##################################### | 100% \u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "minimap2-2.28        | 1.2 MB    | 4                                     |   1% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "minimap2-2.28        | 1.2 MB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "kernel-headers_linux | 921 KB    | 6                                     |   2% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "dashing-1.0          | 2.9 MB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "sysroot_linux-64-2.1 | 14.5 MB   | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "coverm-0.7.0         | 2.2 MB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "coverm-0.7.0         | 2.2 MB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "samtools-1.21        | 476 KB    | #2                                    |   3% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "kernel-headers_linux | 921 KB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "samtools-1.21        | 476 KB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "strobealign-0.15.0   | 280 KB    | ##1                                   |   6% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "starcode-1.4         | 39 KB     | ###############1                      |  41% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "strobealign-0.15.0   | 280 KB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "starcode-1.4         | 39 KB     | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "fastani-1.34         | 133 KB    | ####4                                 |  12% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "bwa-0.7.18           | 193 KB    | ###                                   |   8% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "fastani-1.34         | 133 KB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "bwa-0.7.18           | 193 KB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "htslib-1.21          | 3.0 MB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\n",
      "k8-1.2               | 7.2 MB    | ##################################### | 100% \u001b[A\n",
      "\n",
      "openblas-0.3.28      | 5.5 MB    | ##################################### | 100% \u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "minimap2-2.28        | 1.2 MB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "minimap2-2.28        | 1.2 MB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "dashing-1.0          | 2.9 MB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "coverm-0.7.0         | 2.2 MB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "gsl-2.7              | 3.2 MB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "samtools-1.21        | 476 KB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "samtools-1.21        | 476 KB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "strobealign-0.15.0   | 280 KB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "strobealign-0.15.0   | 280 KB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "starcode-1.4         | 39 KB     | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "starcode-1.4         | 39 KB     | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "fastani-1.34         | 133 KB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "fastani-1.34         | 133 KB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "bwa-0.7.18           | 193 KB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "bwa-0.7.18           | 193 KB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "kernel-headers_linux | 921 KB    | ##################################### | 100% \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "                                                                                \u001b[A\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\n",
      "\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "                                                                                \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
      "donearing transaction: - \n",
      "donefying transaction: - \n",
      "doneuting transaction: - \n",
      "\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "#conda install -c bioconda -c conda-forge coverm "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "8c32a215-2be6-42d8-83da-ab5ce32899ad",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Submitted batch job 38736\n"
     ]
    }
   ],
   "source": [
    "THREADS = 14\n",
    "logdir = f'{LOGS}/CoverM'\n",
    "\n",
    "todo = '''\n",
    "outdir={OUTDIR}\n",
    "mkdir -p $outdir\n",
    "\n",
    "echo \"Processing {BATCH}\"\n",
    "sras=({IDS})\n",
    "for sra in ${{sras[@]}}; do\n",
    "    out=$outdir/${{sra}}.txt\n",
    "    CR1=$CR/${{sra}}_1.fq.gz\n",
    "    CR2=$CR/${{sra}}_2.fq.gz\n",
    "\n",
    "    #skip processed files\n",
    "    if [ -f $out ]; then\n",
    "        echo \"$sra already exists. Skipping.\"\n",
    "        continue\n",
    "    else\n",
    "        echo \"Processing $sra\"\n",
    "        coverm contig --methods rpkm -t {THREADS} -c $CR1 $CR2 -r $PS/representative_viral.fasta -o $out \\\n",
    "            --min-read-percent-identity 90 --min-read-aligned-percent 75 --min-covered-fraction 75\n",
    "    fi\n",
    "done\n",
    "echo \"All done, check the outputs\"\n",
    "'''\n",
    "\n",
    "for host in hosts:\n",
    "    if host not in ['turkey']:\n",
    "        continue\n",
    "        \n",
    "    IDS = ' '.join([i for i in meta.loc[meta.isolation_host == host].index]) # if i in cleans])\n",
    "    binac2(tasks=THREADS, time='60:00:00', mem=32, name=f'{host}_CoverM', logdir=logdir, \n",
    "           todo=todo.format(BATCH=host, IDS=IDS, OUTDIR=f'{PS}/CoverM', THREADS=THREADS))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 181,
   "id": "b970e8ac-4c0f-40ea-a9ee-5ff42a789530",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!rm -r {PS}/CoverM/CoverM_mapped.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "19ce8078-4927-4307-8936-0d61e26f5d9c",
   "metadata": {},
   "outputs": [],
   "source": [
    "#collect mapping states\n",
    "\n",
    "coverm_mapped = pd.DataFrame()\n",
    "\n",
    "for log in os.listdir('PS/Logs/CoverM'):\n",
    "    if log.startswith('.'):\n",
    "        continue\n",
    "    elif log.endswith('.error'):\n",
    "        host = log.split('_')[0]\n",
    "        with open(f'PS/Logs/CoverM/{log}') as error:\n",
    "            maps = [m.split('/CoverM/')[-1] for m in error.read().split(' total (')[:-1] if 'Skipping' not in m]\n",
    "        for mp in maps:\n",
    "            sample = mp.split('.txt')[0].strip()\n",
    "            mapped = mp.split('found ')[-1].split(' reads')[0]\n",
    "            total = mp.split('out of ')[-1].strip()\n",
    "            coverm_mapped.loc[sample, ['Host', 'Mapped', 'Total']] = host, int(mapped), int(total)\n",
    "\n",
    "coverm_mapped = coverm_mapped.loc[coverm_mapped['Total'] != 0]\n",
    "coverm_mapped['Perc_mapped'] = coverm_mapped.Mapped*100/coverm_mapped.Total\n",
    "coverm_mapped['Perc_mapped'] = coverm_mapped['Perc_mapped'].apply(lambda x: round(x, 3))\n",
    "coverm_mapped.sort_values(by=['Host', 'Perc_mapped'], ascending=[True, False], inplace=True)\n",
    "\n",
    "coverm_mapped.to_csv('PS/CoverM_mapped.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "46a11396-4e12-40fd-bb04-eb422e6ae95e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "239"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(coverm_mapped)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "8b164adb-481d-4e04-bafa-fbc738e99802",
   "metadata": {},
   "outputs": [],
   "source": [
    "#collect rpkm values\n",
    "\n",
    "dfs = []\n",
    "for txt in os.listdir('PS/CoverM'):\n",
    "    if txt.startswith('.') or os.path.getsize(f'PS/CoverM/{txt}') == 0:\n",
    "        continue\n",
    "    elif txt.endswith('.txt'):\n",
    "        df = pd.read_csv(f'PS/CoverM/{txt}', sep='\\t', index_col=0)\n",
    "        df.columns = [c.split('/')[-1].split('_')[0] for c in df.columns]\n",
    "        dfs.append(df)\n",
    "\n",
    "coverm_rpkm = pd.concat(dfs, axis=1)        \n",
    "coverm_rpkm.to_csv('PS/CoverM_RPKM.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "001172bc-7eb3-4e78-a966-4c80a2d818aa",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "243"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(os.listdir('PS/CoverM'))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f2a9b33e-b328-47af-bc48-cfda1f55f9c0",
   "metadata": {},
   "source": [
    "# Status"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "37ccd8f1-b81c-494a-9230-a1c8cb3a9b24",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Fri Jan 24 16:00:35 2025\n",
      "             JOBID PARTITION     NAME     USER    STATE       TIME TIME_LIMI  NODES NODELIST(REASON)\n"
     ]
    }
   ],
   "source": [
    "!squeue -l -u ho_graaf20"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "id": "0adcd312-caea-492c-a3cf-5de9352091ac",
   "metadata": {},
   "outputs": [],
   "source": [
    "!scancel 38632"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b1f0429c-ffc6-4795-bf40-89e21c00d78c",
   "metadata": {},
   "outputs": [],
   "source": [
    "#\n",
    "!scontrol show job 38626"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "ea0957e5-0149-4a35-9177-d5d3fa782654",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "JobId=38626 JobName=cow_QC\n",
      "   UserId=ho_graaf20(900741) GroupId=ho_genetics(500055) MCS_label=N/A\n",
      "   Priority=7000 Nice=0 Account=bw18d010 QOS=normal\n",
      "   JobState=COMPLETED Reason=None Dependency=(null)\n",
      "   Requeue=1 Restarts=0 BatchFlag=1 Reboot=0 ExitCode=0:0\n",
      "   RunTime=00:00:01 TimeLimit=10-00:00:00 TimeMin=N/A\n",
      "   SubmitTime=2025-01-20T13:30:52 EligibleTime=2025-01-20T13:30:52\n",
      "   AccrueTime=2025-01-20T13:30:52\n",
      "   StartTime=2025-01-20T13:30:53 EndTime=2025-01-20T13:30:54 Deadline=N/A\n",
      "   PreemptEligibleTime=2025-01-20T13:30:53 PreemptTime=None\n",
      "   SuspendTime=None SecsPreSuspend=0 LastSchedEval=2025-01-20T13:30:53 Scheduler=Main\n",
      "   Partition=compute AllocNode:Sid=login02.cluster:100071\n",
      "   ReqNodeList=(null) ExcNodeList=(null)\n",
      "   NodeList=node2-092\n",
      "   BatchHost=node2-092\n",
      "   NumNodes=1 NumCPUs=8 NumTasks=8 CPUs/Task=1 ReqB:S:C:T=0:0:*:*\n",
      "   ReqTRES=cpu=8,mem=32G,node=1,billing=520\n",
      "   AllocTRES=cpu=8,mem=32G,node=1,billing=520\n",
      "   Socks/Node=* NtasksPerN:B:S:C=8:0:*:* CoreSpec=*\n",
      "   MinCPUsNode=8 MinMemoryNode=32G MinTmpDiskNode=0\n",
      "   Features=(null) DelayBoot=00:00:00\n",
      "   OverSubscribe=OK Contiguous=0 Licenses=(null) Network=(null)\n",
      "   Command=/pfs/10/project/bw18d010/ho_graaf20/ChickenPhages/Logs/QC/cow_QC.sh\n",
      "   WorkDir=/home/ho/ho_genetics/ho_graaf20/MyProjects/ChickenPhages\n",
      "   StdErr=/pfs/10/project/bw18d010/ho_graaf20/ChickenPhages/Logs/QC/cow_QC.error\n",
      "   StdIn=/dev/null\n",
      "   StdOut=/pfs/10/project/bw18d010/ho_graaf20/ChickenPhages/Logs/QC/cow_QC.log\n",
      "   \n",
      "\n"
     ]
    }
   ],
   "source": [
    "!scontrol show job 38626"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "9cfa78a0-8cbb-4c4a-8634-3b50809c18f3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "sbatch -t -p {queue} --mem={mem}g -J {name} --output={logdir} --error={logdir} --ntasks-per-node={tasks} -N {nodes} {script}\n"
     ]
    }
   ],
   "source": [
    "time = 1 \n",
    "!echo sbatch -t $time -p {queue} --mem={mem}g -J {name} --output={logdir} --error={logdir} \\\n",
    "     --ntasks-per-node={tasks} -N {nodes} {script}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "aeaa5868-dc11-4f85-ae43-75d84f9fc3fd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Usage: ln [OPTION]... [-T] TARGET LINK_NAME\n",
      "  or:  ln [OPTION]... TARGET\n",
      "  or:  ln [OPTION]... TARGET... DIRECTORY\n",
      "  or:  ln [OPTION]... -t DIRECTORY TARGET...\n",
      "In the 1st form, create a link to TARGET with the name LINK_NAME.\n",
      "In the 2nd form, create a link to TARGET in the current directory.\n",
      "In the 3rd and 4th forms, create links to each TARGET in DIRECTORY.\n",
      "Create hard links by default, symbolic links with --symbolic.\n",
      "By default, each destination (name of new link) should not already exist.\n",
      "When creating hard links, each TARGET must exist.  Symbolic links\n",
      "can hold arbitrary text; if later resolved, a relative link is\n",
      "interpreted in relation to its parent directory.\n",
      "\n",
      "Mandatory arguments to long options are mandatory for short options too.\n",
      "      --backup[=CONTROL]      make a backup of each existing destination file\n",
      "  -b                          like --backup but does not accept an argument\n",
      "  -d, -F, --directory         allow the superuser to attempt to hard link\n",
      "                                directories (note: will probably fail due to\n",
      "                                system restrictions, even for the superuser)\n",
      "  -f, --force                 remove existing destination files\n",
      "  -i, --interactive           prompt whether to remove destinations\n",
      "  -L, --logical               dereference TARGETs that are symbolic links\n",
      "  -n, --no-dereference        treat LINK_NAME as a normal file if\n",
      "                                it is a symbolic link to a directory\n",
      "  -P, --physical              make hard links directly to symbolic links\n",
      "  -r, --relative              create symbolic links relative to link location\n",
      "  -s, --symbolic              make symbolic links instead of hard links\n",
      "  -S, --suffix=SUFFIX         override the usual backup suffix\n",
      "  -t, --target-directory=DIRECTORY  specify the DIRECTORY in which to create\n",
      "                                the links\n",
      "  -T, --no-target-directory   treat LINK_NAME as a normal file always\n",
      "  -v, --verbose               print name of each linked file\n",
      "      --help     display this help and exit\n",
      "      --version  output version information and exit\n",
      "\n",
      "The backup suffix is '~', unless set with --suffix or SIMPLE_BACKUP_SUFFIX.\n",
      "The version control method may be selected via the --backup option or through\n",
      "the VERSION_CONTROL environment variable.  Here are the values:\n",
      "\n",
      "  none, off       never make backups (even if --backup is given)\n",
      "  numbered, t     make numbered backups\n",
      "  existing, nil   numbered if numbered backups exist, simple otherwise\n",
      "  simple, never   always make simple backups\n",
      "\n",
      "Using -s ignores -L and -P.  Otherwise, the last option specified controls\n",
      "behavior when a TARGET is a symbolic link, defaulting to -P.\n",
      "\n",
      "GNU coreutils online help: <https://www.gnu.org/software/coreutils/>\n",
      "Full documentation <https://www.gnu.org/software/coreutils/ln>\n",
      "or available locally via: info '(coreutils) ln invocation'\n"
     ]
    }
   ],
   "source": [
    "!ln --help"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f7e9f2b6-8a8d-4d8e-aa4b-b3c450db3d02",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
